src/urllib3/response.py

```diff
@@ -280,6 +280,22 @@ def get(self, n: int) -> bytes:
 
         return ret.getvalue()
 
+    def get_all(self) -> bytes:
+        buffer = self.buffer
+        if not buffer:
+            assert self._size == 0
+            return b""
+        if len(buffer) == 1:
+            result = buffer.pop()
+        else:
+            # BytesIO.writelines says "newlines are not added"
+            with io.BytesIO() as fd:
+                fd.writelines(buffer)
+                result = fd.getvalue()
```

Comment:
Can this be simplified?
```suggestion
            result = b"".join(buffer)
```

Reply:
not sure why that didn't occur to me! think I was basing it in `get` and didn't notice that's all it was doing in the end